source/common/upstream/cluster_manager_impl.cc

```diff
@@ -768,14 +761,31 @@ ClusterManagerImpl::ThreadLocalClusterManagerImpl::ClusterEntry::connPool(
     return nullptr;
   }
 
+  // Inherit socket options from downstream connection, if set.
+  uint32_t hash_key = 0;
```

Comment:
Use `Optional<uint32_t>` to avoid maintaining 2 variables.

Reply:
Changed.